src/Microsoft.AspNetCore.Mvc.Core/Internal/ResponseCacheFilterExecutor.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.ResponseCaching;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class ResponseCacheFilterExecutor
+    {
+        private readonly CacheProfile _cacheProfile;
+        private int? _cacheDuration;
+        private ResponseCacheLocation? _cacheLocation;
+        private bool? _cacheNoStore;
+        private string _cacheVaryByHeader;
+        private string[] _cacheVaryByQueryKeys;
+
+        public ResponseCacheFilterExecutor(CacheProfile cacheProfile)
+        {
+            _cacheProfile = cacheProfile ?? throw new ArgumentNullException(nameof(cacheProfile));
+        }
+
+        public int Duration
```

Comment:
Rather than duplicating all of this, it would be better pass in the `ResponseCacheFilter` object directly - similar to how our action result executors work.